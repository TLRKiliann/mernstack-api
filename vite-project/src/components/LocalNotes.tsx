import { Component } from 'react'
import {notesType} from '../Models/notesType'
import SubLocalComp from './location/SubLocalComp'
import DerivedLocalNotes from './location/DerivedLocalNotes'

interface childrenProps {
  location: string | undefined
  children: React.ReactNode
}

export default class LocalNotes extends Component<notesType, childrenProps> {
  constructor(props:any) {
    super(props)
    this.state = {
      location: this.props.location
    }
    this.handleReduce = this.handleReduce.bind(this);
  }

  handleReduce = () => {
    if (this.state.location.length >= 1) {
      const toSlice = this.state.location.slice(0, 1);
      this.setState({location: toSlice});
    }
    else {
      this.setState({location : "ok"});
    }
  }
  render() {
    console.log("localNotes:", this.state.location)
    return(
      <div style={{margin: '10px', border: '1px solid yellow'}}>
        <label>1 - Reusable BTN ! {this.props.location}</label>
        <button
          onClick={this.handleReduce}
          style={{marginLeft: '10px'}}
        >
          Abbr location
        </button>

        <SubLocalComp subLocation={this.state.location}/>

        <DerivedLocalNotes>
          {this.props.children}
        </DerivedLocalNotes>
      </div>
    )
  }
}