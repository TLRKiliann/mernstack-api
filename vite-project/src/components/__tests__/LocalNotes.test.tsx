//import React from "react";
import {beforeEach, describe, expect, test, it, vi} from 'vitest';
import {fireEvent, render, screen, waitFor} from '@testing-library/react';
//import "@testing-library/jest-dom/extend-expect"
//import "@testing-library/jest-dom";
import { act } from 'react-dom/test-utils';
import {create} from 'react-test-renderer';
import LocalNotes from "../LocalNotes.tsx";

test('MatchSnapShot test LocalNotes 1', () => {
	const tree = create(<LocalNotes />)
	expect(tree.toJSON()).toMatchSnapshot();
})