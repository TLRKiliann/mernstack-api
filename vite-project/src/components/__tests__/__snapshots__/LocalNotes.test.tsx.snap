// Vitest Snapshot v1

exports[`MatchSnapShot test LocalNotes 1 1`] = `
<div>
  <label>
    1 - Hello LocalNote ! 
  </label>
  <button
    onClick={[Function]}
    style={
      {
        "marginLeft": "10px",
      }
    }
  >
    Abbr location
  </button>
  <div
    style={
      {
        "border": "1px solid lightgreen",
        "color": "lightgreen",
      }
    }
  >
    2 - SubComponent subLocation: 
  </div>
  <div />
</div>
`;
